core/src/types/address.rs

```diff
@@ -576,18 +578,35 @@ pub fn masp_tx_key() -> crate::types::key::common::SecretKey {
     common::SecretKey::try_from_slice(bytes.as_ref()).unwrap()
 }
 
+/// Temporary helper for testing, a hash map of tokens addresses with their
+/// informal currency codes and number of decimal places.
+pub fn tokens() -> HashMap<Address, (&'static str, Denomination)> {
+    vec![
+        (nam(), ("NAM", 6.into())),
+        (btc(), ("BTC", 8.into())),
+        (eth(), ("ETH", 18.into())),
+        (dot(), ("DOT", 10.into())),
+        (schnitzel(), ("Schnitzel", 6.into())),
+        (apfel(), ("Apfel", 6.into())),
+        (kartoffel(), ("Kartoffel", 6.into())),
+    ]
+    .into_iter()
+    .collect()
+}
+
 /// Temporary helper for testing, a hash map of tokens addresses with their
 /// MASP XAN incentive schedules. If the reward is (a, b) then a rewarded tokens
 /// are dispensed for every b possessed tokens.
-pub fn masp_rewards() -> HashMap<Address, (u64, u64)> {
+pub fn masp_rewards() -> HashMap<(Address, Option<Key>, MaspDenom), (u64, u64)>
+{
     vec![
-        (nam(), (0, 100)),
-        (btc(), (1, 100)),
-        (eth(), (2, 100)),
-        (dot(), (3, 100)),
-        (schnitzel(), (4, 100)),
-        (apfel(), (5, 100)),
-        (kartoffel(), (6, 100)),
+        ((nam(), None, MaspDenom::Zero), (0, 100)),
```

Comment:
The rewards are equal across denominations, so we don't need to return a denomination here.